kakarot_scripts/utils/kakarot.py

```diff
@@ -386,27 +388,38 @@ async def get_eoa(private_key=None, amount=10) -> Account:
     )
 
 
-async def send_pre_eip155_transaction(starknet_address, signed_tx: bytes):
+async def send_pre_eip155_transaction(
+    evm_address: str,
+    starknet_address: Union[int, str],
+    signed_tx: bytes,
+    kakarot_owner: Account,
+):
+    rlp_decoded = rlp.decode(signed_tx)
+    v, r, s = rlp_decoded[-3:]
+    unsigned_tx_data = rlp_decoded[:-3]
+    unsigned_encoded_tx = rlp.encode(unsigned_tx_data)
+    tx_hash = int.from_bytes(keccak(unsigned_encoded_tx), "big")
```

Comment:
`msg_hash` ?

Reply:
indeed got a bit confused here, msg_hash is the transaction data without sig, while tx_hash = hash(tx_data, sig)